beets/autotag/hooks.py

```diff
@@ -68,17 +76,45 @@ class AlbumInfo(AttrDict):
     The others are optional and may be None.
     """
 
-    def __init__(self, tracks, album=None, album_id=None, artist=None,
-                 artist_id=None, asin=None, albumtype=None, va=False,
-                 year=None, month=None, day=None, label=None, mediums=None,
-                 artist_sort=None, releasegroup_id=None, catalognum=None,
-                 script=None, language=None, country=None, style=None,
-                 genre=None, albumstatus=None, media=None, albumdisambig=None,
-                 releasegroupdisambig=None, artist_credit=None,
-                 original_year=None, original_month=None,
-                 original_day=None, data_source=None, data_url=None,
-                 discogs_albumid=None, discogs_labelid=None,
-                 discogs_artistid=None, **kwargs):
+    # TYPING: are all of these correct? I've assumed optional strings
+    def __init__(
+            self,
+            tracks: List['TrackInfo'],
+            album: Optional[str] = None,
+            album_id: Optional[str] = None,
+            artist: Optional[str] = None,
+            artist_id: Optional[str] = None,
+            asin: Optional[str] = None,
+            albumtype: Optional[str] = None,
+            va: bool = False,
+            year: Optional[str] = None,
+            month: Optional[str] = None,
+            day: Optional[str] = None,
+            label: Optional[str] = None,
+            mediums: Optional[int] = None,
+            artist_sort: Optional[str] = None,
+            releasegroup_id: Optional[str] = None,
+            catalognum: Optional[str] = None,
+            script: Optional[str] = None,
+            language: Optional[str] = None,
+            country: Optional[str] = None,
+            style: Optional[str] = None,
+            genre: Optional[str] = None,
+            albumstatus: Optional[str] = None,
+            media: Optional[str] = None,
+            albumdisambig: Optional[str] = None,
+            releasegroupdisambig: Optional[str] = None,
+            artist_credit: Optional[str] = None,
+            original_year: Optional[str] = None,
+            original_month: Optional[str] = None,
+            original_day: Optional[str] = None,
+            data_source: Optional[str] = None,
+            data_url: Optional[str] = None,
+            discogs_albumid: Optional[str] = None,
+            discogs_labelid: Optional[str] = None,
+            discogs_artistid: Optional[str] = None,
+            **kwargs,
```

Comment:
This is a heroic set of annotations! We should probably someday drop all these positional parameters and cram everything into `kwargs`. But this will do for now!